rspirv/binary/decoder.rs

```diff
@@ -160,20 +161,31 @@ impl<'a> Decoder<'a> {
     /// null character (`\0`), or reaching the limit or end of the stream
     /// and erroring out.
     pub fn string(&mut self) -> Result<String> {
-        let start_offset = self.offset;
-        let mut bytes = vec![];
-        loop {
-            let word = self.word()?;
-            bytes.extend(&word.to_le_bytes());
-            if bytes.last() == Some(&0) {
-                break;
-            }
-        }
-        while !bytes.is_empty() && bytes.last() == Some(&0) {
-            bytes.pop();
+        // If we have a limit, then don't search further than we need to.
+        let slice = match self.limit {
+            Some(limit) => &self.bytes[self.offset..(self.offset + limit * WORD_NUM_BYTES)],
+            None => &self.bytes[self.offset..],
+        };
+        // Find the null terminator.
+        let first_null_byte =
+            slice
+                .iter()
+                .position(|&c| c == 0)
+                .ok_or_else(|| match self.limit {
+                    Some(_) => Error::LimitReached(self.offset + slice.len()),
+                    None => Error::StreamExpected(self.offset),
```

Comment:
Shouldn't this also be self.offset + slice.len() since previously this would only happen scanning one of the last words in the slice?

Reply:
No, tests assert this behavior. I don't know why, but, they do.